core/promise.js

```diff
@@ -11,16 +11,25 @@
 	if ( window.Promise ) {
 		CKEDITOR.tools.promise = Promise;
 	} else {
-		var polyfillURL = CKEDITOR.getUrl( 'vendor/promise.js' );
+		var polyfillURL = CKEDITOR.getUrl( 'vendor/promise.js' ),
+			isAmdEnv = typeof window.define === 'function' && window.define.amd && typeof window.require === 'function';
+
+		if ( isAmdEnv ) {
+			return window.require( [ polyfillURL ], function( Promise ) {
```

Comment:
From what I can see, `window.require` loads a given script - this means `CKEDITOR.scriptLoader.load` below is not needed for AMD env (and should be probably placed in `else`)?

I wonder about build version, how it will behave there, since the file is bundled (AFAIR) 🤔Could you check it? Will AMD try to load this file separately?

Reply:
> and should be probably placed in `else`

`else` is not necessary as execution is halted on `require` due to `return` statement.

In case of built version, promise file seems to be left unbundled (however it's nicely minified) and it's loaded separately.

Reply:
Ok, somehow missed the `return` statement 🤦‍♂